src/helpers/channels/clone_channel.ts

```diff
@@ -0,0 +1,24 @@
+import { cacheHandlers } from "../../cache.ts";
+import { createChannel } from "./create_channel.ts";
+import { CreateGuildChannel } from "../../types/guilds/create_guild_channel.ts";
+import { DiscordenoChannel } from "../../structures/channel.ts";
+
+/** Create a copy of a channel */
+export async function cloneChannel(channelId: string, reason?: string) {
+  const channelToClone = await cacheHandlers.get(
+    "channels",
+    channelId
+  );
+  //Return undefined if channel is not cached (unsure about error handling)
+  if (!channelToClone) throw new Error(Errors.CHANNEL_NOT_FOUND);
+
+  //If "name" is undefined as specified by types
+  channelToClone.name ??= "new-channel";
```

Comment:
1. Add a check to make sure dm channels don't work
2. Remove this name thing as it will always have a name